src/imperative/cached_op.cc

```diff
@@ -126,6 +126,8 @@ CachedOp::CachedOp(
         fwd_graph_.outputs.push_back(nodeEntry);
       }
     }
+    fwd_graph_ = exec::EliminateCommonExpr(std::move(fwd_graph_));
+
```

Comment:
any unit test?

Reply:
Frankly I'm not sure how to properly test it - any suggestions?

Reply:
Maybe we can test after `FHasDeterministicOutput` property is added to some specific operator?